includes/Core/User_Input/User_Input.php

```diff
@@ -0,0 +1,314 @@
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of REST_Route objects.
+	 */
+	private function get_rest_routes() {
+		$can_authenticate = function() {
+			return current_user_can( Permissions::AUTHENTICATE );
+		};
+
+		return array(
+			new REST_Route(
+				'core/user/data/user-input-settings',
+				array(
+					array(
+						'methods'             => WP_REST_Server::READABLE,
+						'callback'            => function() {
+							return rest_ensure_response( $this->get_answers() );
+						},
+						'permission_callback' => $can_authenticate,
+					),
+					array(
+						'methods'             => WP_REST_Server::CREATABLE,
+						'callback'            => function( WP_REST_Request $request ) {
+							$data = $request->get_param( 'data' );
+
+							if ( ! isset( $data['settings'] ) || ! is_array( $data['settings'] ) ) {
+								return new WP_Error(
+									'rest_missing_callback_param',
+									__( 'Missing settings data.', 'google-site-kit' ),
+									array( 'status' => 400 )
+								);
+							}
+
+							return rest_ensure_response(
+								$this->set_answers(
+									$data['settings']
+								)
+							);
+						},
+						'permission_callback' => $can_authenticate,
+						'args'                => array(
+							'data' => array(
+								'type'       => 'object',
+								'required'   => true,
+								'properties' => array(
+									'settings' => array(
+										'type'      => 'object',
+										'required'  => true,
+										'questions' => array_fill_keys(
+											array_keys( static::$questions ),
+											array(
+												'type'  => 'array',
+												'items' => array( 'type' => 'string' ),
+											)
+										),
+									),
+								),
+							),
+						),
+					),
+				)
+			),
+		);
+	}
+
+	/**
+	 * Gets the set of user input questions.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array The user input questions.
+	 */
+	public static function get_questions() {
+		return static::$questions;
+	}
+
+	/**
+	 * Gets user input answers.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array|WP_Error User input answers.
+	 */
+	public function get_answers() {
+		$questions = static::$questions;
+		$settings  = array_merge(
+			$this->site_specific_questions->get(),
+			$this->user_specific_questions->get()
+		);
+
+		// If there are no settings, return default empty values.
+		if ( empty( $settings ) ) {
+			array_walk(
+				$questions,
+				function ( &$question ) {
+					$question['values'] = array();
+				}
+			);
+
+			return $questions;
+		}
+
+		$user_id = get_current_user_id();
+
+		foreach ( $settings as &$setting ) {
+			if ( ! isset( $setting['answeredBy'] ) ) {
+				continue;
+			}
+
+			$answered_by = intval( $setting['answeredBy'] );
+			unset( $setting['answeredBy'] );
+
+			if ( ! $answered_by || $answered_by === $user_id ) {
+				continue;
+			}
+
+			$setting['author'] = array(
+				'photo' => get_avatar_url( $answered_by ),
+				'name'  => get_the_author_meta( 'user_email', $answered_by ),
```

Comment:
`get_the_author_meta` is more for front-end use I believe so I don't think that's needed here. Also, the key here is `name` but this is providing their email. Does it need to be `name` or could it be `login` instead like we have elsewhere?

https://github.com/google/site-kit-wp/blob/a1327c4f3c572a326d51fc6fe5e7c695e2859f98/includes/Core/Modules/Modules.php#L1405

Looking at the other similar case we have, I'm surprised to see it also uses the same function 😄 IMO this could be refactored with the following

```php
( new WP_User( $answered_by ) )->user_email
```

[The design shows what looks like a display name](https://www.figma.com/file/CdtrHB0HAjfGRWTnTMMQI0/SummaryWidget?node-id=2074%3A61302&t=gWL4H68yse0QECLp-4) but we should use their username for consistency as this is how we reference other users in other places like notices/errors/DS, etc.

Reply:
Updated to use `login` as key and `( new WP_User( $answered_by ) )->user_login` to show username as instructed. Thanks!